File: src/main/java/com/uiptv/server/HttpManifestServer.java
package com.uiptv.server;

import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import com.uiptv.util.StringUtils;

import java.io.IOException;
import java.nio.file.Path;

import static com.uiptv.util.ServerUtils.generateJsonResponse;

public class HttpManifestServer implements HttpHandler {
    @Override
    public void handle(HttpExchange ex) throws IOException {
        try {
            Path filePath = StaticWebFileResolver.resolve(ex);
            generateJsonResponse(ex, StringUtils.EMPTY + StaticWebFileResolver.readUtf8(filePath));
        } catch (IOException e) {
            ex.sendResponseHeaders(404, -1);
        }
    }
}

File: server/src/main/kotlin/com/uiptv/server/HttpManifestServer.kt
package com.uiptv.server

import com.sun.net.httpserver.HttpExchange
import com.sun.net.httpserver.HttpHandler
import com.uiptv.util.ServerUtils.generateJsonResponse
import com.uiptv.util.StringUtils
import java.io.IOException

class HttpManifestServer : HttpHandler {
    @Throws(IOException::class)
    override fun handle(ex: HttpExchange) {
        try {
            val filePath = StaticWebFileResolver.resolve(ex)
            generateJsonResponse(ex, StringUtils.EMPTY + StaticWebFileResolver.readUtf8(filePath))
        } catch (_: IOException) {
            ex.sendResponseHeaders(404, -1)
        }
    }
}
